src/rust/engine/process_execution/src/lib.rs

```diff
@@ -483,6 +484,27 @@ pub struct ProcessExecutionEnvironment {
     pub strategy: ProcessExecutionStrategy,
 }
 
+#[derive(DeepSizeOf, Debug, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum ProcessConcurrency {
+    /// A range of acceptable cpu cores with optional bounds
+    Range {
+        // Minimum number of cores to use, defaults to 1
+        min: Option<usize>,
+        // Maximum number of cores to use, defaults to min
+        max: Option<usize>,
+    },
+    /// Exclusive access to all cores
+    Exclusive,
+}
+
+impl FromStr for ProcessConcurrency {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        serde_json::from_str(s).map_err(|e| e.to_string())
```

Comment:
Should I be using json here? Is it worth making the cli argument parsing nicer?

Reply:
It might be better to have a more intuitive and succinct cli argument, since that will be hard to deprecate later. OTOH this is only used in the `process_executor` utility binary, not by regular pants users, so maybe not.

On balance, I think I would go for: `M-N` for min=M, max=N where either M or N can be omitted, or a literal `X` for exclusive?

Reply:
How is `process_executor` invoked? Is it a debug tool for people working on pants? Or is it consumed by pants somehow? 

Allowing the omission of the lower bound as a cli argument is a little awkward, but I'm fine with it if you are

`--concurrency=-2`  (min: none, max: 2)
`--concurrency=1`  (min: 1, max: none)
`--concurrency=1-2`  (min: 1, max: 2)

maybe we could use colon instead of dash? 

`--concurrency=:2`
`--concurrency=1`
`--concurrency=1:2`

Reply:
Hmm yeah, maybe `:` is better, with the caveat that in Python the colon indexing syntax is half-open, so `1:2` means `[1]` and so on. Might that be confusing?

Reply:
It's a utility for debugging pants, and rarely used even in that context. TBH it doesn't really matter what its CLI is like, if you want to punt on all this.

Reply:
done. I went with `,` to avoid the python-ism confusion